### PR TITLE
Add Pydantic model support

### PR DESCRIPTION
**Pull request by PeterC-DLS**

### CODE REVIEW

File: api/tests/test_api.py [hunk 1]
```diff
@@ -234,3 +235,53 @@ async def test_push_points():
...
+
+class TrialB(BaseModel):
+    integers: list[int]
+    floats: list[float]
+    original: TrialA
+
+
+@app.post("/test_pydantic")
+@message_unpack
+async def test_pydantic(data: TrialA) -> TrialB:
```
Comment (VictoriaBeilsten-Edmands) [inline api/tests/test_api.py:253]: Does this need marking with `@pytest.mark.asyncio`

Comment (PeterC-DLS) [reply api/tests/test_api.py:253]: No, it's a test endpoint and gets renamed in later PR.

File: api/tests/test_api.py [hunk 2]
```diff
@@ -234,3 +235,53 @@ async def test_push_points():
             assert response.status_code == 200
             assert mp_unpackb(response._content) == "data sent"
         del ps
+
+
+class TrialA(BaseModel):
+    integers: list[int]
```
Comment (VictoriaBeilsten-Edmands) [inline api/tests/test_api.py:241]: Capitalise list type as `List`

Comment (PeterC-DLS) [reply api/tests/test_api.py:241]: New to [Python 3.9](https://docs.python.org/3.11/whatsnew/3.9.html#type-hinting-generics-in-standard-collections) is [PEP 585](https://peps.python.org/pep-0585) that allows type hints in generics with standard collections.

**Review by VictoriaBeilsten-Edmands — APPROVED** [review]

**Pull request merged**